python/metatensor-operations/metatensor/operations/contiguous.py

```diff
@@ -0,0 +1,91 @@
+from . import _dispatch
+from ._backend import (
+    Labels,
+    TensorBlock,
+    TensorMap,
+    torch_jit_is_scripting,
+    torch_jit_script,
+)
+
+def is_contiguous(tensor: TensorMap) -> bool:
+    r"""
+    Checks contiguity of values and gradients in  a :py:class:`TensorMap`
+    
+    :param A: the input :py:class:`TensorMap`.
+
+    :return: a boolean indicating (non-)contiguity of values in ``A``.
+    """
+    check_contiguous = True
+    for key, block in tensor.items():
+        # Here, call another function: def is_contiguous_block(block: TensorBlock) -> bool
+        if not is_contiguous_block(block):
+            check_contiguous = False
+
+        for param, gradient in block.gradients():
+            if not is_contiguous_block(gradient):
+                check_contiguous = False
+
+    return check_contiguous
+
+
+def is_contiguous_block(block: TensorBlock) -> bool:
+    r"""
+    Checks contiguity of values in  a :py:class:`TensorBlock`
+    
+    :param A: the input :py:class:`TensorBlock`.
+
+    :return: a boolean indicating (non-)contiguity of values in ``A``.
+    """
+    check_contiguous = True
+    if not _dispatch.is_contiguous_array(block.values):
+        check_contiguous = False
+
+    return check_contiguous
+
+
+def make_contiguous(tensor: TensorMap) -> TensorMap:
+    r"""
+    Return a new :py:class:`TensorMap` where the values are made to be contiguous.
+
+    :param A: the input :py:class:`TensorMap`.
+
+    :return: a new :py:class:`TensorMap` with the same metadata as ``A`` and
```

Comment:
Both the metadata and data should still be the same. Perhaps we could also write a unit test that checks this? You can use `metatensor.equal(tensor_1, tensor_2)` to check this.
```suggestion
    :return: a new :py:class:`TensorMap` with the same data and metadata as ``tensor`` and
```